src/main/kotlin/exercise3/task1/BalancedBrackets.kt

```diff
@@ -26,7 +26,21 @@ package exercise3.task1
 
 
 internal fun isExpressionBalanced(expression: String): Boolean {
-    TODO("Implement me!!!")
+    val openBrackets = listOf('(', '[', '{')
+    val closedBrackets = listOf(')', ']', '}')
+    var brackets = mutableListOf<Char>()
+
+
+    for(char in expression) {
+        when(char) {
+            in openBrackets -> brackets.add(char)
```

Comment:
The solution is correct. Readability can be improved by using the map as a data structure.